Bluetooth: controller: Fix advertiser filtering in ISR

carlescufi opened the pull request.

carlescufi committed:
Bluetooth: controller: Fix AdvA checks in advertiser

The AdvA in the CONN_IND packet must match the AdvA in the advertising
packet that triggered it regardless of the advertising type.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
Bluetooth: controller: Fix directed adv filtering

The specification states that filter policies shall be ignored for
directed advertising, so reflect this behaviour in the code.
Additionally when the local device is using RPAs but the peer uses an
identity address, the resolving list index needs to be updated when
traversing the RL to reflect that there has indeed been a  device match
even though the IRK match did not happen.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

cvinayak commented on subsys/bluetooth/controller/ll_sw/ctrl.c:
```diff
@@ -728,43 +728,55 @@ static inline bool isr_adv_sr_check(struct pdu_adv *pdu, u8_t devmatch_ok,
-			 ci->payload.connect_ind.adv_addr, BDADDR_SIZE) &&
-		 isr_adv_tgta_check(adv, ci, rl_idx)));
+			 ci->payload.connect_ind.adv_addr, BDADDR_SIZE)) ||
+		 (memcmp(adv->payload.adv_ind.addr,
```
!memcmp ?

cvinayak commented on subsys/bluetooth/controller/ll_sw/ctrl.c:
```diff
@@ -713,13 +713,13 @@ static u32_t isr_rx_adv_sr_report(struct pdu_adv *pdu_adv_rx, u8_t rssi_ready)
-		(devmatch_ok) || (ctrl_irk_whitelisted(rl_idx))) &&
+				 pdu->payload.scan_req.scan_addr, rl_idx)) ||
+		(devmatch_ok) || (ctrl_irk_whitelisted(*rl_idx))) &&
 		(1 /** @todo own addr match check */);
```
can you fix this todo?

carlescufi committed:
Bluetooth: controller: Fix RL permission check

When the initiator is using an RPA and we match it using the filtering,
we should return early and allow the device packet through instead of
going through the whole resolving list.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
Bluetooth: controller: Check AdvA in scan requests

The specification states that the AdvA in a Scan Request packet should
be identical to the one sent in the original advertising packet, so
check this when processing a Scan Request.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>
